maven-plugin/src/main/java/io/github/belgif/rest/guide/validator/maven/plugin/AbstractValidatorMojo.java

```diff
@@ -44,9 +44,15 @@ public abstract class AbstractValidatorMojo extends AbstractMojo {
     /**
      * Output directory for the validation report file
      */
-    @Parameter(property = "rest-guide-validator.outputDir", defaultValue = "target")
+    @Parameter(property = "rest-guide-validator.outputDir", defaultValue = "${project.build.directory}")
     File outputDir;
 
+    /**
+     * Output file for JSON validation report. OutputDir and a default filename will be used when absent.
+     */
+    @Parameter(property = "rest-guide-validator.outputFile")
```

Comment:
Does it work to set the default to `${rest-guide-validator.outputDir}/validationReport.json`? Then it's more clear in maven help and you can leave the logic out of the JsonOutputProcessor.

Reply:
Not possible as you suggested, but I agree that it's better to have the responsibility for the file location in the maven mojo. So modified (simplified) accordingly.